packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -644,21 +646,10 @@ export class ApexDebug extends LoggingDebugSession {
     response: DebugProtocol.LaunchResponse,
     args: LaunchRequestArguments
   ): Promise<void> {
-    if (typeof args.trace === 'boolean') {
-      this.trace = args.trace ? [TRACE_ALL] : undefined;
-      this.traceAll = args.trace;
-    } else if (typeof args.trace === 'string') {
-      this.trace = args.trace.split(',').map(category => category.trim());
-      this.traceAll = this.trace.indexOf(TRACE_ALL) >= 0;
-    }
-    if (this.trace && this.trace.indexOf(TRACE_CATEGORY_PROTOCOL) >= 0) {
-      // only log debug adapter protocol if 'protocol' tracing flag is set, ignore traceAll here
-      logger.setup(Logger.LogLevel.Verbose, false);
-    } else {
-      logger.setup(Logger.LogLevel.Stop, false);
-    }
-
     response.success = false;
+    this.initBreakpointSessionServices(args);
```

Comment:
Moved code into its own methods to make `launchRequest` easier to follow and be in line with what we did in `apexReplayDebug.ts`